### PR TITLE
adi:ad7134: Add support for ad7134

### PR DESCRIPTION
**Pull request by jansunil**
1. Add pyadi driver for ad405x family
2. Add example script for data capture with AD4052
3. Add documentation

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Hardware:
* OS:

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have signed off all commits and they contain "Signed-off by: <insert name>"
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jansunil** [comment]
Changelog: Fixed pylint and build errors

**Comment by jansunil** [comment]
@tfcollins  - Could you please review these changes when you get a chance?

File: supported_parts.md [hunk 1]
```diff
@@ -70,6 +70,8 @@
 - AD5940
 - AD6676
 - AD7124
+- AD7134
+- AD4134
```
Comment (tfcollins) [inline supported_parts.md:74]: Please put these in numeric order

File: adi/ad7134.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
+# Copyright (C) 2024 Analog Devices, Inc.
+#
+# SPDX short identifier: ADIBSD
+# Copyright (C) 2024 Analog Devices, Inc.
```
Comment (tfcollins) [inline adi/ad7134.py:4]: Full license not needed. Just the SPDX

File: adi/ad7134.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+
+class ad7134(rx, context_manager):
+
+    """ AD7134 ADC """
+
+    _complex_data = False
+    channel = []  # type: ignore
```
Comment (tfcollins) [inline adi/ad7134.py:51]: Should be called "channels". Needs to be reset/initialized in the constructor similar to the issue in the related PR

**Review by tfcollins — CHANGES_REQUESTED** [review]
Only minor issues to fix

**Comment by jansunil** [comment]
Changelog: Addressed review comments
1) Corrected the license header
2) reword channel->channels
3) Ordered AD4134 in numeric order in the supported_parts.md

**Comment by jansunil** [comment]
@tfcollins - Do these changes look good now?

**Pull request merged**